src/main/java/ru/hogwarts/school/controller/StudentByCategoryController.java

```diff
@@ -0,0 +1,33 @@
+package ru.hogwarts.school.controller;
+
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import ru.hogwarts.school.model.StudentByCategory;
+import ru.hogwarts.school.service.StudentServiceImpl;
+
+import java.util.List;
+
+@RestController
+@RequestMapping(path = "/student-by-category")
+public class StudentByCategoryController {
+
+    private final StudentServiceImpl studentService;
```

Comment:
```suggestion
    private final StudentService studentService;
```
инъекцию надо всегда делать по интерфейсу, а не реализации